### PR TITLE
Code refactor

### PR DESCRIPTION
**Pull request by d367wang**

### CODE REVIEW

File: src/checkers/inference/dataflow/InferenceTransfer.java [hunk 1]
```diff
@@ -61,8 +61,11 @@
     // case where the correct, inferred RHS has no primary annotation
     private Map<Tree, Pair<RefinementVariableSlot, RefinementVariableSlot>> createdTypeVarRefinementVariables = new HashMap<>();
```
Comment (wmdietl) [inline src/checkers/inference/dataflow/InferenceTransfer.java:62]:
```suggestion
    private final Map<Tree, Pair<RefinementVariableSlot, RefinementVariableSlot>> createdTypeVarRefinementVariables = new HashMap<>();
```

File: src/checkers/inference/dataflow/InferenceTransfer.java [hunk 2]
```diff
@@ -59,10 +59,13 @@
...
     // Type variables will have two refinement variables (one for each bound).  This covers the
     // case where the correct, inferred RHS has no primary annotation
-    private Map<Tree, Pair<RefinementVariableSlot, RefinementVariableSlot>> createdTypeVarRefinementVariables = new HashMap<>();
+    private final Map<Tree, Pair<RefinementVariableSlot, RefinementVariableSlot>> createdTypeVarRefinementVariables = new HashMap<>();
+
+    private final InferenceAnnotatedTypeFactory typeFactory;
 
     public InferenceTransfer(InferenceAnalysis analysis) {
         super(analysis);
+        typeFactory = (InferenceAnnotatedTypeFactory) analysis.getTypeFactory();
```
Comment (d367wang) [inline src/checkers/inference/dataflow/InferenceTransfer.java:68]: check scope of InferenceTransfer

File: src/checkers/inference/dataflow/InferenceTransfer.java [hunk 3]
```diff
@@ -59,10 +59,13 @@
 
     // Type variables will have two refinement variables (one for each bound).  This covers the
     // case where the correct, inferred RHS has no primary annotation
-    private Map<Tree, Pair<RefinementVariableSlot, RefinementVariableSlot>> createdTypeVarRefinementVariables = new HashMap<>();
+    private final Map<Tree, Pair<RefinementVariableSlot, RefinementVariableSlot>> createdTypeVarRefinementVariables = new HashMap<>();
```
Comment (d367wang) [inline src/checkers/inference/dataflow/InferenceTransfer.java:62]: file an issue about `Tree` being the map key

File: src/checkers/inference/dataflow/InferenceTransfer.java (hunk 2 above)
Comment (d367wang) [reply src/checkers/inference/dataflow/InferenceTransfer.java:68]: InferenceTransfer confirmed singleton

File: src/checkers/inference/dataflow/InferenceTransfer.java (hunk 3 above)
Comment (wmdietl) [reply src/checkers/inference/dataflow/InferenceTransfer.java:62]: Filed #374

**Pull request merged**